qcodes/dataset/database_fix_functions.py

```diff
@@ -55,18 +55,32 @@ def fix_version_4a_run_description_bug(conn: ConnectionPlus) -> Dict[str, int]:
         runs_inspected = 0
         runs_fixed = 0
 
+        old_style_keys = ['paramspecs']
+        new_style_keys = ['parameters', 'dependencies', 'inferences',
+                          'standalones']
+
         for run_id in pbar:
 
             desc_str = get_run_description(conn, run_id)
             desc_ser = json.loads(desc_str)
             idps_ser = desc_ser['interdependencies']
 
-            if RunDescriber._is_description_old_style(idps_ser):
+            if list(idps_ser.keys()) == old_style_keys:
                 pass
-            else:
-                new_desc = RunDescriber.from_json(desc_str)
-                update_run_description(conn, run_id, new_desc.to_json())
+            elif list(idps_ser.keys()) == new_style_keys:
+                # we cheat a little here; we introduce a RunDescriber version
+                # number into a DB of schema version 4. That should be okay,
+                # since the only rule is that DBs of schema versions > 4
+                # MUST have a versioned RunDescriber,
+                desc_ser['version'] = 1
```

Comment:
Why is this not version 0

Reply:
Because it's not :)
Slightly less cryptic, the bug that this fixes is that we accidentally wrote version 1 RunDescribers to disk (although they were unversioned). When we read them back, the conversion system can thus correctly convert them if we tag them as being version 1.

Reply:
discussion online sorted this out. I will improve the code and comments so that everything is properly explained and documented.